docs/src/pages/components/data-grid/sorting/ComparatorSortingGrid.tsx

```diff
@@ -1,89 +1,94 @@
 import * as React from 'react';
 import {
   GridColumns,
-  GridRowsProp,
   DataGrid,
-  GridSortDirection,
   GridValueGetterParams,
-  GridSortModel,
+  gridNumberComparator,
+  gridStringOrNumberComparator,
 } from '@mui/x-data-grid';
-import { randomCreatedDate, randomUpdatedDate } from '@mui/x-data-grid-generator';
+import { useDemoData } from '@mui/x-data-grid-generator';
 
-const columns: GridColumns = [
-  { field: 'name' },
-  { field: 'age', type: 'number' },
-  {
-    field: 'username',
-    valueGetter: (params: GridValueGetterParams) =>
-      `${params.row.name || 'unknown'} - ${params.row.age || 'x'}`,
-    sortComparator: (v1, v2, param1, param2) =>
-      (param1.api.getCellValue(param1.id, 'age') as number) -
-      (param2.api.getCellValue(param2.id, 'age') as number),
-    width: 150,
-  },
-  { field: 'dateCreated', type: 'date', width: 180 },
-  { field: 'lastLogin', type: 'dateTime', width: 180 },
-];
+const VISIBLE_FIELDS = ['rating', 'country', 'isAdmin'];
 
-const rows: GridRowsProp = [
-  {
-    id: 1,
-    name: 'Damien',
-    age: 25,
-    dateCreated: randomCreatedDate(),
-    lastLogin: randomUpdatedDate(),
-  },
-  {
-    id: 2,
-    name: 'Nicolas',
-    age: 36,
-    dateCreated: randomCreatedDate(),
-    lastLogin: randomUpdatedDate(),
-  },
-  {
-    id: 3,
-    name: 'Kate',
-    age: 19,
-    dateCreated: randomCreatedDate(),
-    lastLogin: randomUpdatedDate(),
-  },
-  {
-    id: 4,
-    name: 'Sebastien',
-    age: 28,
-    dateCreated: randomCreatedDate(),
-    lastLogin: randomUpdatedDate(),
-  },
-  {
-    id: 5,
-    name: 'Louise',
-    age: 23,
-    dateCreated: randomCreatedDate(),
-    lastLogin: randomUpdatedDate(),
-  },
-  {
-    id: 6,
-    name: 'George',
-    age: 10,
-    dateCreated: randomCreatedDate(),
-    lastLogin: randomUpdatedDate(),
-  },
-];
+interface NameAdminCellValue {
+  name: string;
+  isAdmin: boolean;
+}
 
 export default function ComparatorSortingGrid() {
-  const [sortModel, setSortModel] = React.useState<GridSortModel>([
-    {
-      field: 'username',
-      sort: 'asc' as GridSortDirection,
-    },
-  ]);
+  const { data } = useDemoData({
+    dataSet: 'Employee',
+    visibleFields: VISIBLE_FIELDS,
+    rowLength: 100,
+  });
+
+  const columns = React.useMemo<GridColumns>(
+    () => [
+      {
+        field: 'nameAdmin',
+        headerName: 'Name',
+        valueGetter: (params: GridValueGetterParams) => ({
+          name: params.row.name,
+          isAdmin: params.row.isAdmin,
+        }),
+        renderCell: (params) => {
```

Comment:
Why not using a `valueFormatter`?

Reply:
Done
The typing descriptions are misleading, it was saying that the `value` was not the value post `valueGetter` but it is.
In v6 I think we should clean the all those methods to be extra clear about when it's the raw value and when it is not.